akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/EventSourcedBehavior.scala

```diff
@@ -176,6 +173,24 @@ object EventSourcedBehavior {
    */
   def withSnapshotSelectionCriteria(selection: SnapshotSelectionCriteria): EventSourcedBehavior[Command, Event, State]
 
+  /**
+    * Retention bridges snapshot and journal behavior. This defines the retention criteria.
+    *
+    * @param snapshotEveryNEvents Snapshots are used to reduce playback/recovery times.
+    *                             This defines when a new snapshot is persisted.
+    *
+    * @param keepNSnapshots      After a snapshot is successfully completed,
+    *                             - if 2: retain last maximum 2 *`snapshot-size` events and 3 snapshots
+    *                               (2 old + latest snapshot)
+    *                             - if 0: all events with equal or lower sequence number will not be retained.
+    *
+    * @param deleteMessagesOnSnapshot Opt-in ability to delete older events on successful save of snapshot.
+    *                                 Defaults to disabled.
```

Comment:
This is weird to be a option to me, it's either enabled or not, then what does then `None` mean? If we want to provide a way to only say enable snapshots and keep this many, I think an overload would be better, but to explicitly have to decide to enable deletion seems fine to me.

Also, is it `events` rather than `messages` (I think all user facing typed API talks about events rather)

Reply:
Yes I was doing events but Patrik was talking messages so, that train collided. I agree events is much better.

Reply:
events it is